tests/Agent/IntegrationTests/IntegrationTests/AzureFunction/AzureFunctionHttpTriggerTests.cs

```diff
@@ -33,19 +40,29 @@ protected AzureFunctionHttpTriggerTestsBase(TFixture fixture, ITestOutputHelper
                 },
                 exerciseApplication: () =>
                 {
-                    _fixture.Get("api/httpTriggerFunctionUsingAspNetCorePipeline");
-                    _fixture.Get("api/httpTriggerFunctionUsingAspNetCorePipeline"); // make a second call to verify coldStart is not sent
-                    _fixture.Get("api/httpTriggerFunctionUsingSimpleInvocation"); // invoke an http trigger function that does not use the aspnet core pipeline
+                    if (_testMode == AzureFunctionHttpTriggerTestMode.AspNetCorePipeline)
+                    {
+                        _fixture.Get("api/httpTriggerFunctionUsingAspNetCorePipeline?someParameter=foo");
+                        _fixture.Get("api/httpTriggerFunctionUsingAspNetCorePipeline?someParameter=bar"); // make a second call to verify coldStart is not sent
+                        _fixture.Get("api/httpTriggerFunctionUsingSimpleInvocation"); // invoke an http trigger function that does not use the aspnet core pipeline, even in pipeline test mode
+                    }
+                    else
+                    {
+                        _fixture.Get("api/httpTriggerFunctionUsingSimpleInvocation");
+                        _fixture.Get("api/httpTriggerFunctionUsingSimpleInvocation"); // make a second call to verify coldStart is not sent
+                    }
                     _fixture.AgentLog.WaitForLogLines(AgentLogBase.TransactionSampleLogLineRegex, TimeSpan.FromMinutes(2));
                 }
             );
 
             _fixture.Initialize();
         }
 
-        [Fact]
-        public void Test()
+        [SkippableFact]
+        public void Test_SimpleInvocationMode()
         {
+            Skip.IfNot(_testMode == AzureFunctionHttpTriggerTestMode.SimpleInvocation);
```

Comment:
Is it possible to use a Skip.If here instead of Skip.IfNot? The negation makes it harder for me to read.

Reply:
Or maybe it would be better to change it to something like `Skip.IfNot(IsSimpleInvocationTest(_testMode))`? It's just the negation around a conditional that makes it harder to read.

Reply:
The same approach can be taken for the other skippable facts in this test class.

Reply:
I kind of wish the "skip" was more of a "do if" kind of condition. But I refactored it a bit and it's (maybe) a little easier to read now. 

The alternative would have been to implement an entirely separate test that was largely duplicative of this one.